glue_jupyter/common/state_widgets/viewer_image.py

```diff
@@ -76,4 +77,12 @@ def used_on_axis(i):
             'label': (data.world_component_ids[i].label if data.coords
                       else data.pixel_component_ids[i].label),
             'max': data.shape[i]-1,
+            'unit': (data.get_component(data.world_component_ids[i]).units if data.coords
+                     else ''),
+            'world_value': ("%0.4E" % world_axis(data.coords,
+                                                 data,
+                                                 pixel_axis=data.ndim - 1 - i,
+                                                 world_axis=data.ndim - 1 - i
+                                                 )[self.glue_state.slices[i]] if data.coords
```

Comment:
```suggestion
                                                 )[i] if data.coords
```

`self.glue_state.slices[i]` returns the world axis value of the index `i` (i.e. a float value), which can't be used to slice the `world_axis()` object.

Reply:
``self.glue_state.slices`` should just return the pixel indices of slices, so this is ok as-is. Did you see cases where it looked like ``slices`` didn't return pixel axes?

Reply:
Yeah. Currently, it fails with

```
~/projects/glue-jupyter/glue_jupyter/common/state_widgets/viewer_image.py in _sync_sliders_from_state(self, *not_used)
     75         print([self.glue_state.slices[i] if data.coords else '' for i in range(data.ndim) if not used_on_axis(i)])
     76 
---> 77         self.sliders = [{
     78             'index': i,
     79             'label': (data.world_component_ids[i].label if data.coords

~/projects/glue-jupyter/glue_jupyter/common/state_widgets/viewer_image.py in <listcomp>(.0)
     82             'unit': (data.get_component(data.world_component_ids[i]).units if data.coords
     83                      else ''),
---> 84             'world_value': ("%0.4E" % world_axis(data.coords,
     85                                                  data,
     86                                                  pixel_axis=data.n edim - 1 - i,

IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

and the value of `print([self.glue_state.slices[i] if data.coords else '' for i in range(data.ndim) if not used_on_axis(i)])` gives `[698.0]`.

Reply:
You know what, ignore this. I found the issue!